src/side-navigation/side-navigation.scss

```diff
@@ -170,3 +171,71 @@
     }
   }
 }
+
+@mixin iui-side-navigation-submenu {
+  display: none;
+  visibility: hidden;
+  opacity: 0;
+  width: 0;
+  height: 100%;
+  box-sizing: border-box;
+  padding: 0 $iui-sm;
+  overflow-x: hidden;
+  overflow-y: overlay;
+  resize: horizontal;
+  @include themed {
+    background-color: t(iui-color-background-1);
+    border-right: 1px solid t(iui-color-background-5);
+  }
+
+  &.iui-side-navigation-submenu-visible {
+    display: block;
+    visibility: visible;
+    min-width: $iui-3xl * 2;
+    width: $iui-3xl * 4;
+    max-width: 50vw;
+    opacity: 1;
+  }
+
+  @include iui-transition-group;
+
+  .iui-submenu-header {
+    height: $iui-baseline * 5;
+    display: flex;
+    align-items: center;
+    justify-content: space-between;
+
+    .iui-submenu-header-label {
+      @include iui-text(title);
+      display: flex;
+      align-items: center;
+      overflow: hidden;
+
+      button {
+        flex-shrink: 0;
+      }
+
+      > * {
+        white-space: nowrap;
+        overflow: hidden;
+        text-overflow: ellipsis;
+        user-select: none;
```

Comment:
Are you sure about disabling text selection here?

Reply:
Yeah, this one might be too aggressive, I'll remove it.